### ISSUE TITLE
Jetbrains Licenses for Project Contributors

### ISSUE THREAD

**Issue by emabrey** [welovecoding/editorconfig-netbeans#104]
## Jetbrains Licenses for Project Contributors

I have gone through the process of getting contributors to this project open-source license keys for the Jetbrains All Products Pack (woohoo!🎉). I have received 4️ license keys, normally worth about $1000 total, for **free**! They are intended for use by the publicly listed contributors of this project. One of those contributors is myself, but I need to distribute the other 3 license keys to the other 3 contributors: 

- Benny Neugebauer (@bennyn)
- Michael Koppen (@Yserz)
- Junichi Yamamoto (@junichi11)

If you are one of those three people please contact me at 📧 [emilymabrey93@gmail.com](mailto:emilymabrey93@gmail.com) and I will send you a link  for your license key (you will need to enter or register a Jetbrains account as well). Note that these keys are valid for one year, however they are renewable (though the project needs to email a request 30 days before expiry). The license includes automatic upgrades to the more recent versions of the licensed products as well.

### License Restrictions
To use these open source free license keys there is [a license you must agree to](https://www.jetbrains.com/store/license_opensource.html). It includes a very important restriction: 

> ...[D]o not perform any paid support, consulting or training services for your [open source] project, and... do not distribute paid versions of your [open source] software. 

Basically- **only use the licensed products when you are working on free software.** _This_ license is only to be used only for non-commercial products, so if you wish to work on non-free software you must acquire a separate license that allows you to do that. 

The license terms go on to to say this about both who can use the licensed products and when and where the licensed products can be used: 

> The software may be installed on any number of computers and concurrently used by as many open source project members as the number of licenses provided to the project. 

Essentially the licenses are tied to the project itself and not any specific computer or user. That said however, only listed contributors and the main committers are allowed to use the licenses. If someone leaves the project then they should put their license back into the project's "pool" of licenses. It can then be redistributed to someone else who qualifies for the license.

### Licensed Products
|                 Development Environments                 |                     Development Tools                    |
|:--------------------------------------------------------:|:--------------------------------------------------------:|
| [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea) | [ReSharper](https://www.jetbrains.com/resharper)         |
| [CLion](https://www.jetbrains.com/clion)                 | [ReSharper C++](https://www.jetbrains.com/resharper-cpp) |
| [PhpStorm](https://www.jetbrains.com/phpstorm)           | [dotTrace](https://www.jetbrains.com/profiler)           |
| [PyCharm](https://www.jetbrains.com/pycharm)             | [dotMemory](https://www.jetbrains.com/dotmemory)         |
| [RubyMine](https://www.jetbrains.com/ruby)               | [dotCover](https://www.jetbrains.com/dotcover)           |
| [WebStorm](https://www.jetbrains.com/webstorm)           |                                                          |
| [DataGrip](https://www.jetbrains.com/dbe)                |                                                          |
| [AppCode](https://www.jetbrains.com/objc)                |                                                          |

This is a very useful and otherwise expensive freebie from the Jetbeans company- we should consider giving them a thank you 👍 in the README or otherwise indicating that they are contributing to this open source project.

**Comment by bennyn** [comment]
Hey Emily, I'm speechless... This is great news!! 🎈 🐦

As an avid WebStorm user I could definitely make use of it. 😃 What's needed to claim the license? Just this email to you?

If there are still 3 licenses left, then we can check who really needs / wants one because I am still on a paid Jetbrains plan, so I don't need a license right away.

@ZaLiTHkA has been a very active contributor. Maybe he needs a licence? @markiewb What about you?

**Comment by markiewb** [comment]
@bennyn: Thanks for the thought. I already have an IDEA Ultimate subscription.

**Comment by emabrey** [comment]
Yes if you just email me I can reply with a link for you to click. You click the link and sign-in or register a Jetbrains account, and it handles putting you into the licensed organization automatically. Removal is manual, but adding people is easy. I would've just posted that link here, but it's private info since anyone with a Jetbrains account can use it to add themselves to the org. I could've just removed the trolls but I figured this was smarter/easier lol.

**Comment by emabrey** [comment]
@markiewb Just be clear this includes a lot more than IDEA. You can use the non-IDEA software for open source licensed products and continue using the paid license for IDEA and it won't cause any conflicts. But it is probably a good idea to keep the licenses for people that don't have any non-community IDEA license since this project isn't likely to get a ton of use out of the other licenses. The Meta Programming System is really the only other IDE/Tool that usefully supports Java, but I don't see us needing to define any custom DSLs for this plugin.

**Comment by ZaLiTHkA** [comment]
Wow, thanks so much for the offer @bennyn, but I don't feel I've contributed enough to this project to really justify that.. At least not yet anyway. :)

I love the irony though; getting offered a JetBrains license for helping out with testing a NetBeans plugin.

**Comment by emabrey** [comment]
@ZaLiTHkA well it sounds ironic on the surface, I agree, but generally speaking it can be much easier to develop for a platform without using that platform as part of the development. Since IntelliJ provides a built-in EditorConfig support for it's IDEA Ultimate IDE it also provides us with a reference implementation. I can't speak for everyone else, but personally I use IntelliJ and Netbeans both, depending on exactly what I'm working on. To be clear though, this license isn't an endorsement of IntelliJ over Netbeans- both are great and they both have their place.
